File: server/Topics.Data/Entities/Vote.cs
namespace Topics.Data.Entities;

public class Vote
{
    public int Id { get; set; }
    public int VoterId { get; set; }
    public string Type { get; set; }
    public bool Up { get; set; }

    public User Voter { get; set; }
}

public class CommentVote : Vote
{
    public int CommentId { get; set; }

    public Comment Comment { get; set; }
}

public class PostVote : Vote
{
    public int PostId { get; set; }

    public Post Post { get; set; }
}

File: client/core/models/api/vote.ts
import { Comment } from './comment';
import { Post } from './post';
import { User } from './user';

export interface Vote {
  id: number;
  voterId: number;
  type: string;
  up: boolean;

  voter?: User;
}

export interface CommentVote extends Vote {
  commentId: number;
  comment?: Comment;
}

export interface PostVote extends Vote {
  postId: number;
  post?: Post;
}
